ludwig/visualize.py

```diff
@@ -217,19 +221,23 @@ def _extract_ground_truth_values(
 
     :return pd.Series: ground truth values from source data set
     """
-    # determine ground truth data format and get appropriate reader
-    data_format = figure_data_format_dataset(ground_truth)
-    if data_format not in CACHEABLE_FORMATS:
-        raise ValueError(
-            "{} is not supported for ground truth file, " "valid types are {}".format(data_format, CACHEABLE_FORMATS)
-        )
-    reader = get_from_registry(data_format, external_data_reader_registry)
-
-    # retrieve ground truth from source data set
-    if data_format in {"csv", "tsv"}:
-        gt_df = reader(ground_truth, dtype=None)  # allow type inference
+    if not isinstance(ground_truth, str):
+        gt_df = ground_truth
     else:
-        gt_df = reader(ground_truth)
+        # determine ground truth data format and get appropriate reader
+        data_format = figure_data_format_dataset(ground_truth)
+        if data_format not in CACHEABLE_FORMATS:
+            raise ValueError(
+                "{} is not supported for ground truth file, "
+                "valid types are {}".format(data_format, CACHEABLE_FORMATS)
+            )
+        reader = get_from_registry(data_format, external_data_reader_registry)
+
+        # retrieve ground truth from source data set
+        if data_format in {"csv", "tsv"}:
+            gt_df = reader(ground_truth, dtype=None, df_lib=pd)  # allow type inference
+        else:
+            gt_df = reader(ground_truth, df_lib=pd)
```

Comment:
nit: Consider splitting this block into a separate function, i.e. `ground_truth_df = get_ground_truth_df(ground_truth) -> DataFrame`, since after this point you are working with a fixed df.